src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java

```diff
@@ -1,30 +1,106 @@
 package uk.gov.ons.ctp.response.casesvc.message;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.godaddy.logging.Logger;
 import com.godaddy.logging.LoggerFactory;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.IOException;
+import jodd.util.StringUtil;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.integration.annotation.MessageEndpoint;
-import org.springframework.integration.annotation.ServiceActivator;
-import org.springframework.transaction.annotation.Transactional;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.response.casesvc.config.AppConfig;
 import uk.gov.ons.ctp.response.casesvc.message.sampleunitnotification.SampleUnitParent;
 import uk.gov.ons.ctp.response.casesvc.service.CaseService;
+import uk.gov.ons.ctp.response.casesvc.utility.PubSubEmulator;
 
 /** Receive a new case from the Collection Exercise service. */
-@MessageEndpoint
+@Component
 public class CaseCreationReceiver {
   private static final Logger log = LoggerFactory.getLogger(CaseCreationReceiver.class);
-
+  @Autowired private ObjectMapper objectMapper;
   @Autowired private CaseService caseService;
+  @Autowired AppConfig appConfig;
+
+  /**
+   * To process SampleUnitParents read from pubsub This creates application ready event listener to
+   * provide a active subscription for the new case creation against the receiving SampleUnitParents
+   */
+  @EventListener(ApplicationReadyEvent.class)
+  public void acceptSampleUnit() throws IOException {
```

Comment:
Minor, but maybe this should be something like `createCaseCreationSubscription()` or something that better describes what it's doing?

Reply:
I tried to keep it as it was with the previous caseCreationReceiver and I believe the name suggests what it is doing hence did not change it.

Reply:
That's fine 👍